documentation/book/con-mutual-tls-authentication.adoc

```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-user-operator.adoc
+
+[id='con-mutual-tls-authentication-{context}']
+= Mutual TLS authentication for clients
+
+== What is mutual TLS authentication?
+
+Kafka can be configured to use TLS (transport layer security) to provide encrypted communication between Kafka brokers and clients either with or without _mutual authentication_.
+
+Mutual authentication is when both parties to a communication get proof of the identity of the other party: The broker authenticates the client and the client authenticates the broker.
+
+NOTE: In many common uses of TLS (such as the HTTPS protocol used between a web browser and a web server) the authentication is not mutual: Only one party to the communication gets proof of the identity of the other party.
+
+== When to use mutual TLS authentication for clients
+
+Mutual TLS authentication is a suitable choice for authenticating Kafka clients when:
+
+* The client supports authentication using mutual TLS authentication
+* It is necessary or desirable to use TLS certificates (rather than passwords)
+* You are able to reconfigure and restart client applications periodically (so they do not use certificate which have expired).
```

Comment:
I think this is a point worth drawing attention to, because certificate expiration is something which will definitely happen, and it will happen spontaneously without the user having done anything to the `KafkaUser` or `Kafka` cluster recently. So it's a trade-off the user should be making consciously.

You're right that passwords should be rotated, but since there's no mechanism to allow both old and new passwords for an interim period that is something the user will have to manage for themselves. Their application won't spontaneously break because of a decision they took months before.

Reply:
Even if I accept that, I'm afraid that most of them will not know what it means only from this bullet point. So I'm still not sure how much helpful this is.

Reply:
We will need a concept about that for the docs of the cert renewal. I would still be inclined to mention it here, and link to that concept once we have it. But I can remove this if you want. Thoughts?